namui/src/namui/render/rendering_tree/visit.rs

```diff
@@ -244,6 +252,12 @@ impl RenderingTree {
                             height: bottom - top,
                         };
                     }
+                    SpecialRenderingNode::Scale(scale) => {
+                        bounding_box.x *= scale.x;
+                        bounding_box.y *= scale.y;
+                        bounding_box.width *= scale.x;
+                        bounding_box.height *= scale.y;
+                    }
```

Comment:
Is it okay to do it after rotate it?
In canvas, axis will rotate together on rotating
![image](https://user-images.githubusercontent.com/38313680/173355363-a6c18678-40bf-42f5-8f91-28fd212dbd7d.png)

But this code looks like this
![image](https://user-images.githubusercontent.com/38313680/173355752-d033a3a0-c54a-4ec4-ad34-65e1da46205e.png)

Is this the difference between Namui and Canvas?

Reply:
As you see [line 210](https://github.com/NamseEnt/namseent/pull/209/files/e250b8bd8461828b4e63c3877080ea00001edf67#diff-2dce2d7bc1e94dbad3db40530d42a7112141839460fb4e29a1e21f407a2a3e57R210), when it rotates bounding box, it just expands.

like this.
![image](https://user-images.githubusercontent.com/3580430/173367634-3c939f64-e1e7-45ad-bb53-db0d0fe8f302.png)

Because it's just 'bounding box', not 'bounding path' so I just did it like this.

Reply:
Yes, bounding box will be scaled with fixed axis.
But actually being drawn on canvas will be scaled with rotated axis. Isn't it?

Reply:
The current bounding box of loop is the being drawn on canvas. So it already drawn in rotated axis.

I mean, let's thing about the tree (parent) 1-2-3-4-5 (children).
if 5 is target, 1,2,3,4 is ancestors. I reversed it's order at line [122](https://github.com/NamseEnt/namseent/pull/209/files/e250b8bd8461828b4e63c3877080ea00001edf67#diff-2dce2d7bc1e94dbad3db40530d42a7112141839460fb4e29a1e21f407a2a3e57R122) so it become 4,3,2,1.

The target, 5, has bounding box already. So this code will process with that bounding box.
if 4 is rotate, 4 will rotate 5's bounding box. ok.
then if 3 is scale, it scale rotated bounding box. ok.

you got it?

Reply:
Aaaaah, got it.